cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceEnsurer.java

```diff
@@ -40,17 +44,21 @@
   private final Service serviceDelegate;
 
   @Inject
-  public DefaultNamespaceEnsurer(final NamespaceAdmin namespaceAdmin) {
+  public DefaultNamespaceEnsurer(CConfiguration cConfiguration, final NamespaceAdmin namespaceAdmin) {
+    final boolean createDefaultNamespace = cConfiguration.getBoolean(Constants.Namespace.CREATE_DEFAULT_NAMESPACE);
```

Comment:
We shouldn't create this ensurer instance at all if no need to create default namespace.